### ISSUE TITLE
Allow enforcing a specific precision for BigDecimalField

### ISSUE THREAD

**Issue by Legioth** [vaadin/flow-components#1181]
When using BigDecimalField for currency values, it's quite common that you want to always use the same precision for all values, e.g. two decimals if the currency is € or $.

If the precision is set to 2, then it would have these implications:
* After calling `field.setValue(BigDecimal.ZERO)` or `field.setValue(new BigDecimal("0.001"))`, the string displayed to the user should be `0.00` and `getValue()` should return the corresponding value. It would most likely be most logical to the user if there would only be one value change event in this case, even though it would violate the expectation that there should be a value change event if `getValue()` is no longer the same as the value most recently passed to `setValue()`.
* If the user types e.g. "2" or "2.001" into the field, then the value change event and `getValue()` should be based on `new BigDecimal("2.00")` and the text shown in the field should be changed to `2.00` as soon as convenient but still not so that it interferes with typing in the field.

**Comment by TatuLund** [comment]
> Wouldn't it be easier to just let users use a Formatter for TextField?

Yes. And the most useful approach is to wrap formatter and converter in CustomField, e.g. like follows if you want to have configuration for separator, decimal mark and number of digits in unisyn:

https://gist.github.com/TatuLund/293ad2cc3a2d10448cbd1630b7464ebf

Live demo: https://v-herd.eu/tatulund-addons/decimal-field
